assembly-syntax/src/parser/grammar.lalrpop

```diff
@@ -1258,6 +1258,26 @@ InstWithBitSizeImmediate: Instruction = {
 }
 
 Push: SmallOpsVec = {
+    <l:@L> "push" "." <imm_value:IntOrHexImm> "[" <i:uint> "]" <r:@R> => {
+        smallvec![Op::Inst(Span::new(
+            span!(source_file.id(), l, r),
+            Instruction::PushWordSlice(
+                imm_value,
+                Range { start: i as usize, end: i as usize + 1 }
+            )
+        ))]
+    },
+
+    <l:@L> "push" "." <imm_value:IntOrHexImm> "[" <l_range:uint> "." "." <r_range:uint> "]" <r:@R> => {
```

Comment:
I would suggest we add a `..` token in the lexer, since that will produce more useful error messages if someone does something like `push.FOO[1.2]`.